subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/TransformationNode.java

```diff
@@ -29,43 +29,57 @@
 import org.gradle.api.internal.artifacts.ivyservice.resolveengine.artifact.ResolvedArtifactSet;
 import org.gradle.execution.plan.Node;
 import org.gradle.execution.plan.TaskDependencyResolver;
+import org.gradle.execution.plan.TransformationNodeIdentifier;
 import org.gradle.internal.operations.BuildOperationCategory;
 import org.gradle.internal.operations.BuildOperationContext;
 import org.gradle.internal.operations.BuildOperationDescriptor;
 import org.gradle.internal.operations.BuildOperationExecutor;
 import org.gradle.internal.operations.BuildOperationQueue;
 import org.gradle.internal.operations.RunnableBuildOperation;
+import org.gradle.util.Path;
 
 import java.io.File;
 import java.util.Collections;
 import java.util.Set;
 import java.util.concurrent.atomic.AtomicInteger;
 
-public abstract class TransformationNode extends Node {
-    private static final AtomicInteger ORDER_COUNTER = new AtomicInteger();
+public abstract class TransformationNode extends Node implements TransformationNodeIdentifier {
+    private static final AtomicInteger SEQUENCE = new AtomicInteger();
 
-    private final int order = ORDER_COUNTER.incrementAndGet();
+    private final int uniqueId = SEQUENCE.incrementAndGet();
```

Comment:
uniqueId may be the wrong term here. It's not unique across builds/daemons/machines

Reply:
I was trying to make it consistent with `TaskIdentity` which also has a `uniqueId`.

Reply:
It turns out, there's already an interface called `TransformationIdentity`. 🤔